src/PlatformShFacade.php

```diff
@@ -43,8 +43,13 @@ public function waitFor($projectId, $environmentName, $sha)
         $waitActivity = null;
         foreach ($activities as $activity) {
             if ($activity['type'] == 'environment.push' &&
-                isset($activity['parameters']['new_commit']) &&
-                $activity['parameters']['new_commit'] == $sha) {
+                // Environments are built post-merge so we have to compare the PR head.
```

Comment:
Don't like all these conditions. Rather:

```php
            if ($activity['type'] == 'environment.push') {
                $env_sha = $this->getShaFromParameters($activity['parameters']);

                if ($env_sha && $env_sha == $sha) {
                    $waitActivity = $activity;
                    break;
                }
            }
```

and:

```php
    /**
     * Get environment sha from activity parameters.
     *
     * @param array $parameters
     *   Parameters from activity.
     *
     * @return string|null
     *   The sha, or null if none found.
     */
    protected function getShaFromParameters($parameters)
    {
        if (isset($parameters['github-pr-head'])) {
            // Environments are built post-merge so we have to compare the PR head.
            return $parameters['github-pr-head'];
        } elseif (isset($parameters['new_commit'])) {
            // Environments are built as normal so we compare the new commit.
            return $parameters['new_commit'];
        }

        return null;
    }
```

Reply:
:+1: I refactored this accordingly with a few additional changes.

I opted for passing the activity object instead of the parameters array to the sha extraction method. This way we place the responsibility for determining which sha to use for an activity on the method whatever structure the activity might have.